sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -737,4 +756,23 @@ class AnalysisSuite extends AnalysisTest with Matchers {
       CollectMetrics("evt1", count :: Nil, tblB))
     assertAnalysisError(query, "Multiple definitions of observed metrics" :: "evt1" :: Nil)
   }
+
+  test("Analysis exceed max iterations") {
+    // RuleExecutor only throw exception or log warning when the rule is supposed to run
+    // more than once.
+    val maxIterations = 2
+    val conf = new SQLConf().copy(SQLConf.ANALYZER_MAX_ITERATIONS -> maxIterations)
```

Comment:
I think its ok to inline the analyzer like this;
```
  test("Analysis exceed max iterations") {
    // RuleExecutor only throw exception or log warning when the rule is supposed to run
    // more than once.
    val maxIterations = 2
    val testAnalyzer = new Analyzer(
      new SessionCatalog(new InMemoryCatalog, FunctionRegistry.builtin, conf),
      new SQLConf().copy(SQLConf.ANALYZER_MAX_ITERATIONS -> maxIterations))

    val plan = testRelation2.select(
      $"a" / Literal(2) as "div1",
      $"a" / $"b" as "div2",
      $"a" / $"c" as "div3",
      $"a" / $"d" as "div4",
      $"e" / $"e" as "div5")

    val message = intercept[TreeNodeException[LogicalPlan]] {
      testAnalyzer.execute(plan)
    }.getMessage
    assert(message.startsWith(s"Max iterations ($maxIterations) reached for batch Resolution"))
  }
```

Reply:
Yeah, I see. It's more concise.